### PR TITLE
[GLUTEN-2098][CH] Refactor HiveTableScanExecTransformer

### PR DESCRIPTION
**Pull request by exmy**
## What changes were proposed in this pull request?

close: #2098 
refactor the code related to `HiveTableScanExecTransformer`

## How was this patch tested?

under test

### CODE REVIEW

File: backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHMetricsApi.scala [hunk 1]
```diff
@@ -53,6 +53,9 @@ class CHMetricsApi extends MetricsApi with Logging with LogLevelUtil {
       "extraTime" -> SQLMetrics.createTimingMetric(sparkContext, "extra operators time")
     )
 
+  override def genBatchScanTransformerMetricsUpdater(
+      metrics: Map[String, SQLMetric]): MetricsUpdater = new BatchScanMetricsUpdater(metrics)
+
```
Comment (KevinyhZou) [inline backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHMetricsApi.scala:58]: why make this change ? it seems not necessary

Comment (exmy) [reply backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHMetricsApi.scala:58]: make `genBatchScanTransformerMetricsUpdater` and `genBatchScanTransformerMetrics` closer in proximity.

**Comment by taiyang-li** [comment]
LGTM

**Review by zzcclp — DISMISSED** [review]
LGTM

File: gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala [hunk 2]
```diff
@@ -1,5 +1,4 @@
 /*
- * Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (rui-mo) [inline gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala:2]: Is this change needed?

Comment (exmy) [reply gluten-core/src/main/scala/org/apache/spark/sql/hive/HiveTableScanExecTransformer.scala]: It was accidentally change, have restored.

**Review by rui-mo — APPROVED** [review]

**Comment by zzcclp** [comment]
Please create a  issue to record this pr.
